### PR TITLE
Feature/nick e adpsfc sfcshp pythonapi

### PR DESCRIPTION
**Pull request by nicholasesposito**
## Description

Provide a detailed description of what this PR does. What problem does it fix? What new capability does it add?

Provides 2 python scripts. Both use the new python API provided by Ron to produce IODA files.
The issues [#66](https://github.com/NOAA-EMC/JEDI-T2O/issues/66) and [#70](https://github.com/NOAA-EMC/JEDI-T2O/issues/70)

NOTE: These are version 1 scripts. Version 2 will be added at a later date.

## Dependencies
#1239 - Ctests are failing because `import bufr`  does not work. Once this PR is closed, I'll re-try the ctest and go from there

## Impact

None

## Checklist
- [x] Add files to repository
- [ ] Get tests to pass

### CODE REVIEW

**Comment by PatNichols** [comment]
@nicholasesposito There's have a fix in my PR for your PR.

**Comment by PatNichols** [comment]
@nicholasesposito My PR is taking longer than expected. If you can grab my version of Ron's python bindings file,
ssrc/bufr/BufrParser/Query/python_bindings.cpp 
. it should fix the issue.

**Comment by nicholasesposito** [comment]
@PatNichols I merged your file into my branch but I'm still getting errors where it can't find bufr or ioda when importing

**Comment by PatNichols** [comment]
@nicholasesposito There was change in ioda which ioda-converters. After the update to your branch I will take a look.

**Comment by nicholasesposito** [comment]
Ok! let me check.

Sincerely,
Nick

-----
Nicholas Esposito (he/him/his)
SAIC, Scientific Programmer / Analyst
***@***.***/NCEP/EMC
5830 University Research Court, Room 2058
College Park, MD 20740

Email: ***@***.***
Office Phone: (301)683-3706

File: test/testinput/prepbufr_adpsfc_api.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+
+import sys
+import numpy as np
+import numpy.ma as ma
+import pyiodaconv.ioda_conv_engines as iconv
+import pyiodaconv.ioda_conv_ncio as iconio
+#import bufr
```
Comment (PatNichols) [inline test/testinput/prepbufr_adpsfc_api.py:11]: Hi @nicholasesposito This should now be 
from pyiodaconv import bufr

File: test/testinput/prepbufr_sfcshp_api.py [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+import sys
+sys.path.append('/work2/noaa/da/nesposito/ioda-bundle_featurequery/build/lib/')
+#Still need these until PR 1239 is merged
+#https://github.com/JCSDA-internal/ioda-converters/pull/1239
+sys.path.append('/work2/noaa/da/nesposito/ioda-bundle_featurequery/build/lib/pyiodaconv/')
+sys.path.append('/work2/noaa/da/nesposito/ioda-bundle_featurequery/build/lib/python3.9/pyioda/')
+
+import numpy as np
+import numpy.ma as ma
+import bufr
```
Comment (PatNichols) [inline test/testinput/prepbufr_sfcshp_api.py:15]: from pyiodaconv import bufr

**Comment by nicholasesposito** [comment]
The tests now pass but only if I have the sys.append.path commands still in there.

**Comment by BenjaminRuston** [comment]
@nicholasesposito you still need the PYTHONPATH appended with a built `pyiodaconv`

we do this as we get into our JEDI environment. may need to ask @CoryMartin-NOAA, but think we should be able to move this forward

File: test/CMakeLists.txt [hunk 3]
```diff
@@ -1328,6 +1333,22 @@ if(iodaconv_bufr_ENABLED)
                             gdas.t12z.adpsfc.tm00.nc ${IODA_CONV_COMP_TOL_ZERO}
                     DEPENDS bufr2ioda.x )
 
+  ecbuild_add_test( TARGET  test_iodaconv_prepbufr_ncep_api_adpsfc2ioda
```
Comment (PatNichols) [inline test/CMakeLists.txt:1336]: At some point in this test definition, you need
ENVIRONMENT "PYTHONPATH=${IODACONV_PYTHONPATH}"
see some of the other tests in this files. That will eliminate the need for appending path.

**Comment by PraveenKumar-NOAA** [comment]
@nicholasesposito **try using these lines in your python api:** 
import pyiodaconv.ioda_conv_engines as iconv
import pyiodaconv.ioda_conv_ncio as iconio
from pyiodaconv import bufr
from pyioda import ioda

**and in your CMakeLists.text something like this:** 
  ecbuild_add_test( TARGET  test_iodaconv_bufr_ncep_prepbufr_adpupa_api
                    TYPE    SCRIPT
                    ENVIRONMENT "PYTHONPATH=${IODACONV_PYTHONPATH}"
                    COMMAND bash
                    ARGS    ${CMAKE_BINARY_DIR}/bin/iodaconv_comp.sh
                            netcdf
                            "${Python3_EXECUTABLE} testinput/bufr_ncep_prepbufr_adpupa.py
                            -i testinput/ADPUPA.prepbufr
                            -o testrun/prepbufr_adpupa_api.nc"
                            prepbufr_adpupa_api.nc ${IODA_CONV_COMP_TOL_ZERO} )

I used these in my api and CMakeLists.text  and my ctest passed successfully. 

For more details look at my PR#1285.

File: test/testinput/prepbufr_adpsfc_api.py (hunk 1 above)
Comment (PatNichols) [reply test/testinput/prepbufr_adpsfc_api.py]: @nicholasesposito I pushed up the changes for the new directory structure. It passes tests on my ubuntu aws instance.

**Comment by PatNichols** [comment]
@nicholasesposito  I committed the changes needed. hope you don't mind.

**Comment by nicholasesposito** [comment]
Yea, no problem. I had another different deadline this week but I'll take a look at this tomorrow morning.

**Review by PatNichols — APPROVED** [review]
Looks like it passing ctests. We might need someone with experience on how this effects things down stream.

**Comment by PatNichols** [comment]
@nicholasesposito looks like this PR is ready for review status but that is your call.

**Comment by nicholasesposito** [comment]
Open for review.
Just FYI version 2 python scripts are nearly complete. A few different variables are on those, but I'll get those in at a later date.

**Comment by BenjaminRuston** [comment]
@emilyhcliu think this one is ready to go now

I'm rebuilding and will test tomorrow just to be thorough, if you have a chance to give it a thumbs up please do

**Review by BenjaminRuston — APPROVED** [review]
these are passing for me locally 1.4.0 stack

**Review by srherbener — APPROVED** [review]
Thanks for adding these tests!

**Pull request merged**